src/main/java/group1/mips_simulator/Utility.java

```diff
@@ -0,0 +1,75 @@
+package group1.mips_simulator;
+
+
+public class Utility {
+    public static final int WORD_SIZE = 16;
+
+    public static boolean isNumeric(String str) {
+        try {
+            Integer.parseInt(str);
+            return true;
+        } catch (NumberFormatException e) {
+            return false;
+        }
+    }
+
+    public static String octalToBinary(String octal) {
+        StringBuilder result = new StringBuilder();
+        for (char c : octal.toCharArray()) {
+            switch (c) {
+                case '0' -> result.append("000");
+                case '1' -> result.append("001");
+                case '2' -> result.append("010");
+                case '3' -> result.append("011");
+                case '4' -> result.append("100");
+                case '5' -> result.append("101");
+                case '6' -> result.append("110");
+                case '7' -> result.append("111");
+                default -> throw new IllegalArgumentException("Octal number contains invalid character: " + octal);
+            }
+        }
+        return result.toString();
+    }
+
+    /**
+     * Convert the octal string number into a binary number representation of the
+     * target size. 0s will be padded to hit the target, or only the 'targetSize'
+     * least significant bits will be returned.
+     * IE: The most significant bits will cut in that case.
+     */
+    public static String octalToBinary(String octal, int targetSize) {
```

Comment:
minor thing that wont affect anything
consider renaming "octal" variable to "octalString" or something, just to be explicit about what value is being passed into some of these functions

Reply:
Ok! I consider the types of the params to be part of the function signature so adding the word "String" seems redundant to me. But that's just a minor style thing so I'll add it now :)

Now if we were in a weakly typed language like python I'd 100% agree with you.

Reply:
thats fair, thats fair